src/app/cli/src/init/coda_run.ml

```diff
@@ -435,53 +435,64 @@ let setup_local_server ?(client_trustlist = []) ?rest_server_port
+              Server.respond_string ~status:`Not_found "Route not found"
+              >>| lift ))
+    |> Deferred.map ~f:(fun _ ->
+           [%log info]
+             !"Created %s at: http://localhost:%i/graphql"
+             server_description port )
+  in
   Option.iter rest_server_port ~f:(fun rest_server_port ->
       trace_task "REST server" (fun () ->
-          let graphql_callback =
-            Graphql_cohttp_async.make_callback
-              (fun _req -> coda)
-              Mina_graphql.schema
-          in
-          Cohttp_async.(
-            Server.create_expert
-              ~on_handler_error:
-                (`Call
-                  (fun _net exn ->
-                    [%log error]
-                      "Exception while handling REST server request: $error"
-                      ~metadata:
-                        [ ("error", `String (Exn.to_string_mach exn))
-                        ; ("context", `String "rest_server") ] ))
-              (Tcp.Where_to_listen.bind_to
-                 (if insecure_rest_server then All_addresses else Localhost)
-                 (On_port rest_server_port))
-              (fun ~body _sock req ->
-                let uri = Cohttp.Request.uri req in
-                let status flag =
-                  let%bind status = Mina_commands.get_status ~flag coda in
-                  Server.respond_string
-                    ( status |> Daemon_rpcs.Types.Status.to_yojson
-                    |> Yojson.Safe.pretty_to_string )
-                in
-                let lift x = `Response x in
-                match Uri.path uri with
-                | "/graphql" ->
-                    [%log debug] "Received graphql request. Uri: $uri"
-                      ~metadata:
-                        [ ("uri", `String (Uri.to_string uri))
-                        ; ("context", `String "rest_server") ] ;
-                    graphql_callback () req body
-                | "/status" ->
-                    status `None >>| lift
-                | "/status/performance" ->
-                    status `Performance >>| lift
-                | _ ->
-                    Server.respond_string ~status:`Not_found "Route not found"
-                    >>| lift ))
-          |> Deferred.map ~f:(fun _ ->
-                 [%log info]
-                   !"Created GraphQL server at: http://localhost:%i/graphql"
-                   rest_server_port ) ) ) ;
+          create_graphql_server
+            ~bind_to_address:
+              Tcp.Bind_to_address.(
+                if insecure_rest_server then All_addresses else Localhost)
+            ~schema:Mina_graphql.schema ~server_description:"GraphQL server"
+            rest_server_port ) ) ;
+  (*Second graphql server with limited queries exopsed*)
+  Option.iter limited_graphql_port ~f:(fun rest_server_port ->
+      trace_task "Second REST server (with limited queries)" (fun () ->
+          create_graphql_server ~bind_to_address:Tcp.Bind_to_address.Localhost
```

Comment:
Can you include the `insecure_rest_server` ternary here? We should probably introduce a new flag to control this one independently of the other one too (sadly). Maybe we can call it `open_limited_graphql_port`?